### ISSUE TITLE
[Parser] .NET accepts different formats for DateTime

### ISSUE THREAD

**Issue by almedina-ms** [Microsoft/AdaptiveCards#1915]
# Platform

* .NET HTML
* .NET WPF
* Android
* iOS
* JavaScript
* UWP

# Author or host

Developer

# Version of SDK

master

# Issue

The .NET parser accepts datetime formats without time and with extra miliseconds

```
{
	"$schema": "http://adaptivecards.io/schemas/adaptive-card.json",
	"type": "AdaptiveCard",
	"version": "1.0",
	"body": [
		{
			"type": "TextBlock",
			"text": "Created {{DATE(2017-02-14,SHORT)}} {{TIME(2017-02-14T06:08:39.5454560000Z)}}",
			"wrap": true
		},
		{
			"type": "TextBlock",
			"text": "Created {{DATE(06:08:39Z,LONG)}} {{TIME(2017-02-14T06:08:39Z)}}",
			"wrap": true
		}
	]
}
```
# .NET
![image](https://user-images.githubusercontent.com/35784165/45580207-7cbed180-b843-11e8-8078-a56407dcfa7d.png)

# UWP and Visualizer
![image](https://user-images.githubusercontent.com/35784165/45580223-96601900-b843-11e8-97aa-ccce90851e16.png)

![image](https://user-images.githubusercontent.com/35784165/45580236-b1328d80-b843-11e8-8685-515caa1624bd.png)

**Comment by khouzam** [comment]
This should have been taken care of by #1874. @carlos-zamora can you verify?

**Comment by carlos-zamora** [comment]
@khouzam Yup. Fixed by the new regex in #1874.
